runtime/lua/vim/ui/float.lua

```diff
@@ -0,0 +1,214 @@
+local Border = require('vim.ui.border')
+local tbl = require('vim.ui.ui_table')
+
+_AssociatedBufs = {}
+
+
+local clear_buf_on_leave = function(bufnr)
+  vim.cmd(
+    string.format(
+      "autocmd WinLeave,BufLeave,BufDelete <buffer=%s> ++once ++nested lua require('vim.ui.float').clear(%s)",
+      bufnr,
+      bufnr
+    )
+  )
+end
+
+local win_float = {}
+
+win_float.default_options = {
+  winblend = 15,
+  percentage = 0.9,
+}
+
+function win_float.default_opts(options)
+  options = tbl.apply_defaults(options, win_float.default_options)
+
+  local width = math.floor(vim.o.columns * options.percentage)
+  local height = math.floor(vim.o.lines * options.percentage)
+
+  local top = math.floor(((vim.o.lines - height) / 2) - 1)
+  local left = math.floor((vim.o.columns - width) / 2)
+
+  local opts = {
+    relative = 'editor',
+    row      = top,
+    col      = left,
+    width    = width,
+    height   = height,
+    style    = 'minimal'
+  }
+
+  return opts
+end
+
+function win_float.centered(options)
+  options = tbl.apply_defaults(options, win_float.default_options)
+
+  local win_opts = win_float.default_opts(options)
+
+  local bufnr = vim.fn.nvim_create_buf(false, true)
+  local win_id = vim.fn.nvim_open_win(bufnr, true, win_opts)
+
+  vim.cmd('setlocal nocursorcolumn')
+  vim.fn.nvim_win_set_option(win_id, 'winblend', options.winblend)
+
+  vim.cmd(
+    string.format(
+      "autocmd WinLeave <buffer> silent! execute 'bdelete! %s'",
+      bufnr
+    )
+  )
+
```

Comment:
This should be changed to `vim.api.nvim_buf_delete` now I think, since that's been implemented since I wrote it.